File: ruby/http-server.rb
require 'webrick'

option = {
  BindAddress: '0.0.0.0',
  Port: '3000',
}
server = WEBrick::HTTPServer.new(option)
logger = server.config[:Logger]

server.mount_proc('/') do |req, res|
  logger.info "Request Header: #{req.header}"
  logger.info "Request Body: #{req.body}"
  res.body = req.body
end

Signal.trap(:INT) { server.shutdown }

server.start

File: deno/http-server.ts
import { serve, Response } from "https://deno.land/std@0.53.0/http/server.ts";

const port = 8000;
const s = serve({ port });
console.log(`listen on http://localhost:${port}/`);

for await (const req of s) {
  const res: Response = {
    body: req.body,
  };
  req.respond(res);
}
